compile_state/src/state.rs

```diff
@@ -1071,10 +1074,111 @@ Example:
 (test::assert-true (ok? nil))
 "#),
             ret: add_special(vm, "return", ""),
+            ns: add_special(vm, "ns", r#"Usage: (ns SYMBOL)
+
+Changes to namespace.  This is "open-ended" change and is intended for use with
+the REPL prefer with-ns for scripts.
+THe symbol "::" will return to the "root" namespace (i.e. no namespace prepended to globals).
+This will cause all globals defined to have namespace:: prepended.
```

Comment:
all globals? or all non globals?

why would globals have `namespace::` prepended to them after using the `::` symbol?

Reply:
Namespaces only apply to globals, locals live on the stack with limited scope so they would not matter.
Yes all globals defined in a namespace get namespace:: prepended.  This means you can not define something outside your namespace but you can "extend" it with embedded ::'s.  Lookups should be fine, it tries the prepended version as well as the non-prepended version.  Maybe allowing :: with def needs some more thought.